identifier: CVE-2019-14235
title:  Denial-of-service possibility in django.utils.text.Truncator
description: "If django.utils.text.Truncator's chars() and words() methods were passed the html=True argument, they were extremely slow to evaluate certain inputs due to a catastrophic backtracking vulnerability in a regular expression"
date: "2019-08-02"
affected_range: '>=1.11 <1.11.23 || >=2.1 <2.1.11 || >=2.2 <2.2.4'
fixed_versions:
- 1.11.23
- 2.1.11
- 2.2.4
affected_versions: 1.x and 2.x series
solution: Upgrade and patches available (see source)
credit: MITRE
urls:
- https://docs.djangoproject.com/en/dev/releases/security/
- https://lists.opensuse.org/opensuse-security-announce/2019-08/msg00006.html
- https://seclists.org/bugtraq/2019/Aug/15	
- https://www.debian.org/security/2019/dsa-4498	
- https://www.djangoproject.com/weblog/2019/aug/01/security-releases/
package_slug: pypi/Django
